addons/account/static/src/js/mail_activity.js

```diff
@@ -1,68 +1,53 @@
-odoo.define('account.activity', function (require) {
-"use strict";
+/** @odoo-module **/
 
-var AbstractField = require('web.AbstractField');
-var core = require('web.core');
-var field_registry = require('web.field_registry');
+import { registry } from "@web/core/registry";
+import { useService } from "@web/core/utils/hooks";
 
-var QWeb = core.qweb;
-var _t = core._t;
+const { Component } = owl;
 
-var VatActivity = AbstractField.extend({
-    className: 'o_journal_activity_kanban',
-    events: {
-        'click .see_all_activities': '_onOpenAll',
-        'click .see_activity': '_onOpenActivity',
-    },
-    init: function () {
+export class VatActivity extends Component {
+    setup() {
+        this.action = useService("action");
         this.MAX_ACTIVITY_DISPLAY = 5;
-        this._super.apply(this, arguments);
-    },
-    //------------------------------------------------------------
-    // Private
-    //------------------------------------------------------------
-    _render: function () {
-        var info = JSON.parse(this.value);
-        if (!info) {
-            this.$el.html('');
-            return;
-        }
-        info.more_activities = false;
-        if (info.activities.length > this.MAX_ACTIVITY_DISPLAY) {
-            info.more_activities = true;
-            info.activities = info.activities.slice(0, this.MAX_ACTIVITY_DISPLAY);
+        this.formatData(this.props);
+    }
+
+    formatData(props) {
+        this.info = JSON.parse(this.props.value);
+        this.info.more_activities = false;
+        if (this.info.activities.length > this.MAX_ACTIVITY_DISPLAY) {
+            this.info.more_activities = true;
+            this.info.activities = this.info.activities.slice(0, this.MAX_ACTIVITY_DISPLAY);
         }
-        this.$el.html(QWeb.render('accountJournalDashboardActivity', info));
-    },
+    }
 
-    _onOpenActivity: function(e) {
-        e.preventDefault();
-        var self = this;
-        self.do_action({
+    async _onOpen(activity) {
+        this.action.doAction({
             type: 'ir.actions.act_window',
-            name: _t('Journal Entry'),
+            name: this.env._t('Journal Entry'),
             target: 'current',
-            res_id: $(e.target).data('resId'),
+            res_id: activity.res_id,
             res_model:  'account.move',
             views: [[false, 'form']],
         });
-    },
+    }
 
-    _onOpenAll: function(e) {
+    _onOpenAll(e) {
         e.preventDefault();
```

Comment:
you can use the .prevent event handler suffix top preventDefault directly in xml